File: src/backend/schemas/solver.py
from typing import Optional

from pydantic import BaseModel


# Shared properties
class SolverBase(BaseModel):
    name: str
    description: str


class SolverCreate(SolverBase):
    pass


class SolverInDBBase(SolverBase):
    id: Optional[int] = None

    class Config:
        orm_mode = True


# Additional properties to return via API
class Solver(SolverInDBBase):
    pass


class SolverData(BaseModel):
    layout_id: int

File: src/frontend/src/app/_dtos/solver.ts
export interface ISolver {
	id: number;
	name: string;
	description: string;
}

export class Solver implements ISolver {
	id: number;
	name: string;
	description: string;

	constructor(solver: ISolver) {
		this.id = solver.id;
		this.name = solver.name;
		this.description = solver.description;
	}
}
